modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java

```diff
@@ -0,0 +1,718 @@
+            ".HTTPBinding=HTTP-Redirect";
+    private static final String USER_AGENT = "Apache-HttpClient/4.2.5 (java 1.5)";
+
+    private static final String PRIMARY_IS_SP_NAME = "travelocity";
+    private static final String PRIMARY_IS_SP_INBOUND_AUTH_TYPE_SAMLSSO = "samlsso";
+    private static final String PRIMARY_IS_SP_AUTHENTICATION_TYPE = "federated";
+
+    private static final String PRIMARY_IS_SAML_ISSUER_NAME = "travelocity.com";
+    private static final String PRIMARY_IS_SAML_ACS_URL = "http://localhost:8490/travelocity.com/home.jsp";
+    private static final String PRIMARY_IS_SAML_NAME_ID_FORMAT = "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress";
+
+    private static final String PRIMARY_IS_IDP_NAME = "trustedIdP";
+    private static final String PRIMARY_IS_IDP_AUTHENTICATOR_NAME_OIDC = "OpenIDConnectAuthenticator";
+    private static final String PRIMARY_IS_IDP_CALLBACK_URL = "https://localhost:9853/commonauth";
+
+    private static final String SECONDARY_IS_TEST_USERNAME = "testFederatedUser";
+    private static final String SECONDARY_IS_TEST_PASSWORD = "testFederatePassword";
+    private static final String SECONDARY_IS_TEST_USER_ROLES = "admin";
+
+    private static final String SECONDARY_IS_SP_NAME = "secondarySP";
+    protected OauthAdminClient adminClient;
+    private String secondaryISClientID;
+    private String secondaryISClientSecret;
+
+    private static final int PORT_OFFSET_0 = 0;
+    private static final int PORT_OFFSET_1 = 1;
+
+    CookieStore cookieStore = new BasicCookieStore();
+    private CloseableHttpClient client;
+//private static final String HTTP_RESPONSE_HEADER_LOCATION = "location";
+
+    @BeforeClass(alwaysRun = true)
+    public void initTest() throws Exception {
+
+        super.initTest();
+
+        super.createServiceClients(PORT_OFFSET_0, sessionCookie,
+                new IdentityConstants.ServiceClientType[]{
+                        IdentityConstants.ServiceClientType.APPLICATION_MANAGEMENT,
+                        IdentityConstants.ServiceClientType.IDENTITY_PROVIDER_MGT,
+                        IdentityConstants.ServiceClientType.SAML_SSO_CONFIG});
+
+        super.createServiceClients(PORT_OFFSET_1, null,
+                new IdentityConstants.ServiceClientType[]{
+                        IdentityConstants.ServiceClientType.APPLICATION_MANAGEMENT,
+                        IdentityConstants.ServiceClientType.OAUTH_ADMIN});
+
+        createServiceProviderInSecondaryIS();
+        createIdentityProviderInPrimaryIS();
+        createServiceProviderInPrimaryIS();
+
+        client = HttpClientBuilder.create().build();
+
+        boolean userCreated = addUserToSecondaryIS();
+        Assert.assertTrue(userCreated, "User creation failed in secondary IS.");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void endTest() throws Exception {
+        try {
+            super.deleteSAML2WebSSOConfiguration(PORT_OFFSET_0, PRIMARY_IS_SAML_ISSUER_NAME);
```

Comment:
no need to delete saml config separately. Deleting the SP will delete this config as well.